connector/src/main/java/org/geysermc/connector/network/translators/java/window/JavaSetSlotTranslator.java

```diff
@@ -86,8 +89,24 @@ public void translate(GeyserSession session, ServerSetSlotPacket packet) {
                 session.getPlayerInventory().setItem(packet.getSlot(), newItem, session);
                 InventoryTranslator.PLAYER_INVENTORY_TRANSLATOR.updateSlot(session, session.getPlayerInventory(), packet.getSlot());
             } else {
-                inventory.setItem(packet.getSlot(), newItem, session);
-                translator.updateSlot(session, inventory, packet.getSlot());
+                if(packet.getWindowId() == 0 && packet.getSlot() == 5) {
+                    CompoundTag profile = null;
+
+                    if (session.getItemMappings().getMapping(newItem.getItemData(session)).getJavaIdentifier().equals("minecraft:player_head")
+                            && newItem.getNbt() != null
+                            && newItem.getNbt().contains("SkullOwner")
```

Comment:
A contains check is not necessary. Generally these just check `get(value) != null` so you're better off just doing the get.